apps/nsbyt/nsbyt.star

```diff
@@ -274,22 +299,38 @@ def getTrip(station_id, station_dest, skiptime):
                     "direction": trip["legs"][0]["direction"],
                     "plannedDateTime": trip["legs"][0]["origin"]["plannedDateTime"],
                     "actualDateTime": originTime,
-                    "actualTrack": trip["legs"][0]["origin"]["plannedTrack"],
+                    "actualTrack": trip["legs"][0]["origin"].get("plannedTrack", "-"),
                     "trainCategory": trip["legs"][0]["product"]["categoryCode"],
+                    "alternativeTransport": trip["legs"][0].get("alternativeTransport"),
+                    "displayName": trip["legs"][0]["product"]["displayName"],
                     "cancelled": trip["legs"][0]["cancelled"],
                 },
             )
 
     return stops
 
 def getTrains(station_id, skiptime):
-    resp = http.get("https://gateway.apiportal.ns.nl/reisinformatie-api/api/v2/departures", params = {"station": station_id}, headers = {"Ocp-Apim-Subscription-Key": API_KEY})
+    # Check the cache for travel info
+    resp_cached = cache.get("ns_%s" % station_id)
```

Comment:
Would be better to use the http module's built in cache functionality. Just pass `ttl_seconds` to `http.get()`.